assert_test.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2023 Snowflake Computing Inc. All rights reserved.
+
+package gosnowflake
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func assertNilF(t *testing.T, actual any, descriptions ...string) {
+	errMsg := validateNil(actual, descriptions...)
+	if errMsg != "" {
+		t.Fatal(errMsg)
+	}
+}
+
+func assertNilE(t *testing.T, actual any, descriptions ...string) {
+	errMsg := validateNil(actual, descriptions...)
+	if errMsg != "" {
+		t.Error(errMsg)
+	}
+}
+
+func assertNotNilF(t *testing.T, actual any, descriptions ...string) {
+	errMsg := validateNotNil(actual, descriptions...)
+	if errMsg != "" {
+		t.Fatal(errMsg)
+	}
+}
+
+func assertNotNilE(t *testing.T, actual any, descriptions ...string) {
+	errMsg := validateNotNil(actual, descriptions...)
+	if errMsg != "" {
+		t.Error(errMsg)
+	}
+}
+
+func assertEqualF(t *testing.T, actual string, expected string, descriptions ...string) {
+	errMsg := validateEqualStrings(actual, expected, descriptions...)
+	if errMsg != "" {
+		t.Fatal(errMsg)
+	}
+}
+
+func assertEqualE(t *testing.T, actual string, expected string, descriptions ...string) {
+	errMsg := validateEqualStrings(actual, expected, descriptions...)
+	if errMsg != "" {
+		t.Error(errMsg)
+	}
+}
+
+func assertStringContainsF(t *testing.T, actual string, expectedToContain string, descriptions ...string) {
+	errMsg := validateStringContains(actual, expectedToContain, descriptions...)
+	if errMsg != "" {
+		t.Fatal(errMsg)
+	}
+}
+
+func assertStringContainsE(t *testing.T, actual string, expectedToContain string, descriptions ...string) {
+	errMsg := validateStringContains(actual, expectedToContain, descriptions...)
+	if errMsg != "" {
+		t.Error(errMsg)
+	}
+}
+
+func assertHasPrefixF(t *testing.T, actual string, expectedPrefix string, descriptions ...string) {
+	errMsg := validateHasPrefix(actual, expectedPrefix, descriptions...)
+	if errMsg != "" {
+		t.Fatal(errMsg)
+	}
+}
+
+func assertHasPrefixE(t *testing.T, actual string, expectedPrefix string, descriptions ...string) {
+	errMsg := validateHasPrefix(actual, expectedPrefix, descriptions...)
+	if errMsg != "" {
+		t.Error(errMsg)
+	}
+}
+
+func validateNil(actual any, descriptions ...string) string {
```

Comment:
I wonder if it is better to stick them closer to the callee. Maybe we can group asserts and their validation function together?

Reply:
we agreed to leave it as is